Index: budgeting_app/models.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from django.db import models\r\n\r\n# Create your models here.\r\n\r\n# class AccountInfo (models.Model):\r\n#     username = models.CharField(max_length=20)\r\n#     password = models.CharField(max_length=20)\r\n\r\n# class BudgetInfo (models.Model):\r\n#     user_budget = models\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/budgeting_app/models.py b/budgeting_app/models.py
--- a/budgeting_app/models.py	(revision ab7547fe74ef9b01c237c636e121bda2efefb0e4)
+++ b/budgeting_app/models.py	(date 1640298892586)
@@ -1,10 +1,39 @@
 from django.db import models
+from django.contrib.auth import models as authmodel
 
 # Create your models here.
 
-# class AccountInfo (models.Model):
-#     username = models.CharField(max_length=20)
-#     password = models.CharField(max_length=20)
+class Project(models.Model):
+    created_by = models.ForeignKey(authmodel.User, on_delete=models.CASCADE,
+                                   related_name="%(app_label)s_%(class)s_related", related_query_name="%(app_label)s_%(class)ss")
+    project_users = models.ManyToManyField(authmodel.User,  related_name="%(app_label)s_%(class)s_related_users", related_query_name="%(app_label)s_%(class)ss_users")
+    name = models.CharField(max_length=50)
+    desc = models.TextField(null=True)
+    created_at = models.DateTimeField(auto_now=True)
+
+class Budget(models.Model):
+    project_id = models.ForeignKey(Project, on_delete=models.CASCADE)
+    budget = models.DecimalField(decimal_places=3, max_digits=20)
+    created_at = models.DateTimeField(auto_now=True)
+    created_by = models.ForeignKey(authmodel.User, on_delete=models.CASCADE)
 
-# class BudgetInfo (models.Model):
-#     user_budget = models
+class Category(models.Model):
+    name = models.CharField(max_length=50)
+    desc = models.TextField(null=True)
+    type = models.CharField(max_length=50)
+
+class Planned(models.Model):
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
+    amount = models.DecimalField(decimal_places=3, max_digits=20)
+    direction = models.CharField(max_length=50)
+
+class Cashflow(models.Model):
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
+    created_by = models.ForeignKey(authmodel.User, on_delete=models.CASCADE)
+    amount = models.DecimalField(decimal_places=3, max_digits=20)
+    created_at = models.DateTimeField(auto_now=True)
+    name = models.CharField(max_length=50)
+    desc = models.TextField(null=True)
+    direction = models.CharField(max_length=50)
\ No newline at end of file
